specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2023-02-01/Metadata.json

```diff
@@ -0,0 +1,822 @@
+          },
+          {
+            "$ref": "#/parameters/MetadataName"
+          },
+          {
+            "$ref": "#/parameters/MetadataPatch"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/MetadataModel"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "MetadataList": {
+      "description": "List of all the metadata.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of metadata.",
+          "items": {
+            "$ref": "#/definitions/MetadataModel"
+          },
+          "type": "array"
+        },
+        "nextLink": {
+          "description": "URL to fetch the next page of metadata.",
+          "readOnly": true,
+          "type": "string"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "MetadataModel": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
+        }
+      ],
+      "description": "Metadata resource definition.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/metadataProperties",
+          "description": "Metadata properties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "metadataPatch": {
+      "description": "Metadata patch request body.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/metadataPropertiesPatch",
+          "description": "Metadata patch request body",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "metadataContentId": {
+      "description": "Static ID for the content.  Used to identify dependencies and content from solutions or community.  Hard-coded/static for out of the box content and solutions. Can be optionally set for user created content to define dependencies.  If an active content item is made from a template, both will have the same contentId.",
+      "type": "string"
+    },
+    "metadataParentId": {
+      "description": "Full parent resource ID of the content item the metadata is for.  This is the full resource ID including the scope (subscription and resource group)",
+      "type": "string"
+    },
+    "metadataVersion": {
+      "description": "Version of the content.  Default and recommended format is numeric (e.g. 1, 1.0, 1.0.0, 1.0.0.0), following ARM template best practices.  Can also be any string, but then we cannot guarantee any version checks",
+      "type": "string"
+    },
+    "metadataKind": {
+      "type": "string",
```

Comment:
>string [](http://example.com/codeflow?start=15&length=6)

This is not really recommended : 

[https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices](https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices)

String types that should have enums
If a property has a finite set of values (enforced by the service), the string property should be declared as an enum type in the Swagger (using the x-ms-enum annotation) to enable client-side input validation and discovery.

Can you please explain why you are switching from enum to string ?

Reply:
@dw511214992 would this change be considered breaking ? changing a type from enum to string ?

Reply:
Hi, @raosuhas:
 
We don't enforce a finite set of values for the property anymore, although it was enforced in older version. 

the reason is that CAT team can release new kind or sub kind of content at any time, our service shouldn't limit its capability in this regard. 

In backend service, we'll maintain backward compatibility until the pP, PP versions deprecated. but in the meantime the Pr doesn't introduce any breaking change.

Reply:
@JeffreyRichter , can you please suggest if changing from enum to string be a breaking change?

Reply:
Yes, it is breaking because the SDKs will generate different code and customers who use an SDK will be forced to modify their code.

Reply:
Thanks @JeffreyRichter for reviewing.

We haven't released any SDK to customers yet for our preview version. would it be able to get a breaking change approval?

Reply:
But you already shipped some SDKs in the past right? Azures policy is that a customers must be able to adopt a new SDK or service version without requiring the customer make any code changes. If the enum changes to string, then ALL existing SDK customers will have to change their code to adopt the new version.

Reply:
@JeffreyRichter, I am sure that NO SDK has been shipped in the past. we scrutinized our situation prior to proposing the changes.

Reply:
OK, then we can allow the "break" since it won't break anyone.

Reply:
Thank you @JeffreyRichter!

Reply:
Breaking changes are approved.